src/frontend/src/functions/auth.tsx

```diff
@@ -7,49 +7,36 @@ import { api, setApiDefaults } from '../App';
 import { ApiEndpoints } from '../enums/ApiEndpoints';
 import { apiUrl } from '../states/ApiState';
 import { useLocalState } from '../states/LocalState';
-import { useSessionState } from '../states/SessionState';
-
-const tokenName: string = 'inventree-web-app';
+import { fetchGlobalStates } from '../states/states';
 
 /**
  * Attempt to login using username:password combination.
- * If login is successful, an API token will be returned.
- * This API token is used for any future API requests.
+ * If login is successful, a CSRF cookie will be set - which can be used to authenticate future requests.
  */
 export const doBasicLogin = async (username: string, password: string) => {
   const { host } = useLocalState.getState();
-  // const apiState = useServerApiState.getState();
 
   if (username.length == 0 || password.length == 0) {
     return;
   }
 
-  // At this stage, we can assume that we are not logged in, and we have no token
-  useSessionState.getState().clearToken();
-
-  // Request new token from the server
+  // Login to the server
   await axios
     .get(apiUrl(ApiEndpoints.user_token), {
       auth: { username, password },
       baseURL: host,
-      timeout: 2000,
-      params: {
-        name: tokenName
-      }
+      timeout: 2000
```

Comment:
You need to have `withCredentials: true` here, so that the cookies are accepted from the response header. Otherwise the `set-cookie` response headers are ignored.

Ref: https://stackoverflow.com/questions/52549079/does-axios-support-set-cookie-is-it-possible-to-authenticate-through-axios-http